experiment/flake-detector/main.go

```diff
@@ -0,0 +1,220 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"sort"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+type prowJob struct {
+	Type    string `json:"type"`
+	Repo    string `json:"repo"`
+	PullSHA string `json:"pull_sha"`
+	Job     string `json:"job"`
+	State   string `json:"state"`
+}
+
+type jobResults struct {
+	flakeCount  int
+	commitCount int
+	chance      float64
+	jobName     string
+}
+
+func (f jobResults) String() string {
+	return fmt.Sprintf("%d/%d\t(%.2f%%)\t%s", f.flakeCount, f.commitCount, f.chance, f.jobName)
+}
+
+type byChance []jobResults
+
+func (ft byChance) Len() int           { return len(ft) }
+func (ft byChance) Swap(i, j int)      { ft[i], ft[j] = ft[j], ft[i] }
+func (ft byChance) Less(i, j int) bool { return ft[i].chance > ft[j].chance }
+
+type options struct {
+	prowURL string
+	repo    string
+	path    string
+	runOnce bool
+}
+
+func flagOptions() options {
+	o := options{}
+	flag.StringVar(&o.prowURL, "prow-url", "https://deck-ci.svc.ci.openshift.org", "Prow frontend base URL. Required for reading job results")
+	flag.StringVar(&o.repo, "repo", "openshift/origin", "Repository to estimate test flakiness for")
```

Comment:
In the spirit of having prow work across multiple repos, I can switch this to prepend the repo name into the metric and expose flake percentages for all repos configured in prow.